airflow/models/baseoperator.py

```diff
@@ -835,7 +835,7 @@ def __init__(
                 f"'{dag.dag_id if dag else ''}.{task_id}'; received '{trigger_rule}'."
             )
 
-        self.trigger_rule = TriggerRule(trigger_rule)
+        self.trigger_rule: TriggerRule = TriggerRule(trigger_rule)
```

Comment:
Mypy really can't work this out? :o

Reply:
I think it has something to do with either (or a combination of) Enum having a custom metaclass, TriggerRule does multi-inheritance, and one of the parents is a built-in.